Add pipelines permissions (#31996)
DataDog/documentation: content/en/logs/log_configuration/pipelines.md

## Pipeline permissions

Pipelines use [Granular Access Control][7]. This means permissions can be assigned to **roles**, **individual users**, and **teams**, ensuring precise control over pipeline resources.

<div class="alert alert-warning">Granular Access Control is available for organizations that do not already have the scoped RBAC feature flag enabled on piplines.</div>

For individual pipelines, administrators can choose the following edit scopes:

* **Unrestricted**: Any user with the `logs_write_pipelines` permission can modify the pipeline.
* **Restricted pipeline editor**: Editing permissions are limited to specific users, teams, or roles.
* **Processor-only editor**: Users are allowed to edit the pipeline's **processors** (including nested pipelines) but cannot modify the pipeline attributes, such as its filter query or its order in the global pipeline list.

> These permissions are fully manageable programmatically through **API and Terraform**.

Author: ```suggestion
You can manage these permissions programmatically through **API and Terraform**.
```

## Preprocessing

Preprocessing of JSON logs occurs before logs enter pipeline processing. Preprocessing runs a series of operations based on reserved attributes, such as `timestamp`, `status`, `host`, `service`, and `message`. If you have different attribute names in your JSON logs, use preprocessing to map your log attribute names to those in the reserved attribute list.

JSON log preprocessing comes with a default configuration that works for standard log forwarders. To edit this configuration to adapt custom or specific log forwarding approaches:

> 1. Navigate to [Pipelines][8] in the Datadog app and select [Preprocessing for JSON logs][9].

Author: ```suggestion
1. Navigate to [Pipelines][8] in Datadog and select [Preprocessing for JSON logs][9].
```

## Create a pipeline

> 1. Navigate to [Pipelines][8] in the Datadog app.

Author: ```suggestion
1. Navigate to [Pipelines][8] in Datadog.
```

### Integration pipeline library

To see the full list of integration pipelines that Datadog offers, browse the [integration pipeline library][10]. The pipeline library shows how Datadog processes different log formats by default.

{{< img src="logs/processing/pipelines/integration-pipeline-library.mp4" alt="Integration pipeline library" video=true style="width:80%;">}}

> To use an integration pipeline, Datadog recommends installing the integration by configuring the corresponding log `source`. Once Datadog receives the first log with this source, the installation is automatically triggered and the integration pipeline is added to the processing pipelines list. To configure the log source, refer to the corresponding [integration documentation][11].

Author: ```suggestion
To use an integration pipeline, Datadog recommends installing the integration by configuring the corresponding log `source`. After Datadog receives the first log with this source, the installation is automatically triggered and the integration pipeline is added to the processing pipelines list. To configure the log source, see the corresponding [integration documentation][11].
```

It's also possible to copy an integration pipeline using the clone button.

{{< img src="logs/processing/pipelines/clone-pipeline-from-library.mp4" alt="Cloning pipeline from Library" video=true style="width:80%;">}}

## Add a processor or nested pipeline

> 1. Navigate to [Pipelines][8] in the Datadog app.

Author: ```suggestion
1. Navigate to [Pipelines][8] in Datadog.
```

2. Hover over a pipeline and click the arrow next to it to expand processors and nested pipelines.
3. Select **Add Processor** or **Add Nested Pipeline**.

### Processors

A processor executes within a pipeline to complete a data-structuring action. See the [Processors docs][3] to learn how to add and configure a processor by processor type, within the app or with the API.

> See [Parsing dates][12] for more information about parsing a custom date and time format and for information on the `timezone` parameter, which is needed if your timestamps are not in UTC.

Author: ```suggestion
See [Parsing dates][12] to learn about custom date and time formats and the require `timezone` parameter for non-UTC timestamps.
```

## Estimated usage metrics

> Estimated usage metrics are displayed per pipeline - specifically, the volume and count of logs being ingested and modified by each pipeline. There is also a link to the out-of-the-box [Logs Estimated Usage Dashboard][13] from every pipeline where you can view that pipeline's usage metrics in more detailed charts.

Author: ```suggestion
Estimated usage metrics are displayed for each pipeline. This shows the the volume and count of logs being ingested and modified by each pipeline. Every pipeline includes a link to the out-of-the-box [Logs Estimated Usage Dashboard][13]. This dashboard offers detailed charts of the pipeline's usage metrics.
```
